README.md

```diff
@@ -1,278 +1,521 @@
 }
 ```
 
 ### List Collections
 
+The result of this command is an array of collection names.
+
 ```ts
-const list = await client.listCollections();
+> const list = await pinecone.listCollections();
+["collection1", "collection2"]
 ```
 
-Example result:
+## Index operations
+
+Pinecone indexes support operations for working with vector data using operations such as upsert, query, fetch, and delete.
+
+### Targeting an index
+
+To perform data operations on an index, you target it using the `index` method.
 
-```json
-["collection1", "collection2"]
 ```
+const pinecone = new Pinecone()
+const index = pinecone.index('test-index')
 
-## Index operations
+// Now perform index operations
+await index.fetch(['1'])
+```
 
-The Pinecone index operations allow you to perform the following operations instances of `Vector`.
+### Targeting a namespace
 
-A `Vector` is defined as follows:
+By default all data operations take place inside the default namespace of `''`. If you are working with other non-default namespaces, you can target them namespace by chaining a call to `namespace()`.
+
+```ts
+const pinecone = new Pinecone();
+const index = pinecone.index('test-index').namespace('ns1');
+
+// Now perform index operations in the targeted index and namesapce
+await index.fetch(['1']);
+```
+
+If needed, you can check the currently targeted index and namespace by inspecting the `target` property of an index object.
+
+```ts
+const pinecone = new Pinecone();
+const index = pinecone.index('test-index').namespace('ns1');
+
+console.log(index.target); // { index: 'test-index', namespace: 'ns1' }
+```
+
+See [Using namespaces](https://docs.pinecone.io/docs/namespaces) for more information.
+
+### Upsert vectors
+
+Pinecone expects vector data to have the following form:
 
 ```ts
 type Vector = {
   id: string;
-  values: number[];
+  values: Array<number>;
+  sparseValues?: Array<number>;
   metadata?: object;
-  sparseValues: {
-    indices: [15, 30, 11];
-    values: [0.1, 0.2, 0.3];
-  }; // optional sparse values
 };
 ```
 
-After selecting an index to operate on, you can:
+To upsert some vectors, you can use the client like so.
 
-### Upsert vectors
+```ts
+const pinecone = new Pinecone();
+
+// Target an index
+const index = pinecone.index('sample-index');
+
+// Prepare your data
+const vectors = [
+  {
+    id: '1',
+    values: [0.236, 0.971, 0.559],
+  },
+  {
+    id: '2',
+    values: [],
+  },
+];
+
+// Upsert the data into your index
+await index.upsert(vectors);
+```
+
+### Seeing index statistics
+
+When experimenting with data operations, it's sometimes helpful to know how many vectors are stored in each namespace. In that case,
+target the index and use the `describeIndexStats()` command.
 
 ```ts
-const upsertRequest: UpsertRequest = {
-  vectors,
-  namespace,
+> await pinecone.index('example-index').describeIndexStats()
+{
+  namespaces: {
+    '': { vectorCount: 10 }
+    foo: { vectorCount: 2000 },
+    bar: { vectorCount: 2000 }
```

Comment:
weird but i assume this is the actual response?

Reply:
Yep, this is what the API gives.